artifactory/commands/npm/installorci.go

```diff
@@ -480,60 +480,45 @@ func (nca *NpmCommandArgs) prepareDependencies(typeRestriction string) error {
 	}
 
 	// Parse the dependencies json object
-	return jsonparser.ObjectEach(data, func(key []byte, value []byte, dataType jsonparser.ValueType, offset int) error {
+	return jsonparser.ObjectEach(data, func(key []byte, value []byte, dataType jsonparser.ValueType, offset int) (err error) {
 		if string(key) == "dependencies" {
-			err := nca.parseDependencies(value, typeRestriction)
-			if err != nil {
-				return err
-			}
+			err = nca.parseDependencies(value, typeRestriction, []string{nca.packageInfo.BuildInfoModuleId()})
 		}
-		return nil
+		return err
 	})
 }
 
 // Parses npm dependencies recursively and adds the collected dependencies to nca.dependencies
-func (nca *NpmCommandArgs) parseDependencies(data []byte, scope string) error {
-	var transitiveDependencies [][]byte
-	err := jsonparser.ObjectEach(data, func(key []byte, value []byte, dataType jsonparser.ValueType, offset int) error {
-		ver, _, _, err := jsonparser.Get(data, string(key), "version")
-		if err != nil && err != jsonparser.KeyPathNotFoundError {
+func (nca *NpmCommandArgs) parseDependencies(data []byte, scope string, pathToRoot []string) error {
+	return jsonparser.ObjectEach(data, func(key []byte, value []byte, dataType jsonparser.ValueType, offset int) error {
+		depName := string(key)
+		ver, _, _, err := jsonparser.Get(data, depName, "version")
+		if err != nil {
 			return errorutils.CheckError(err)
-		} else if err == jsonparser.KeyPathNotFoundError {
-			log.Warn(fmt.Sprintf("npm dependencies list contains the package '%s' without version information. The dependency will not be added to build-info.", string(key)))
```

Comment:
I suggest that we check what ```npm install``` does, in a case where the package.json includes a depedency without a version, and do the same. I think that returning an error is not the rightr approach, because it'll break existing pipelines.